tests/unittests/imageTest.cpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Base/Image.h"
+
+#include "gtest/gtest.h"
+
+#include <cstdio>
+#include <utility>
+
+using namespace glow;
+
+TEST(Image, writePngImage) {
+  auto range = std::make_pair(0.f, 1.f);
+  Tensor localCopy;
+  bool loadSuccess = !readPngImage(&localCopy, "cat_285.png", range);
+  ASSERT_TRUE(loadSuccess);
```

Comment:
it looks like the test is failing here:
```
../tests/unittests/imageTest.cpp:30: Failure
Value of: loadSuccess
  Actual: false
Expected: true
```

it does not exercise read/write/read path yet...

Reply:
Yeah, that's on purpose. I wanted to know if someone could spot something I was obviously doing wrong before I go all the way of adding the cmake plumbing for image copying.

Reply:
haha, test looks good. One thing I noticed, although have not looked a lot:
`writePNG` uses PNG_COLOR_TYPE_RGB_ALPHA, but `readPNG` could use different mode.

Reply:
`readPNG` does strip out alpha if it exists, so that we always have 3 channels. So that does seem like a bug, should probably use `PNG_COLOR_TYPE_RGB` instead

Reply:
Ah, yeah you're right, it's added back in even though the input tensor has 3 channels.

RE: the image appearing black as mentioned in https://github.com/pytorch/glow/issues/1914, this is because of the range being 0to1. If you use 255.f instead of 1.f (so the scale/bias don't do anything) the cat image looks to be printed correctly, but it's rotated 90 degrees.

Reply:
Ah thanks for the range. That's not obvious why we need to use something different for reading and writing. Regarding the 90 degrees rotation that sounds like there is still a bug.

Reply:
Hi @rdzhabarov, @jfix71,

I have a quick look at the read and write functions and I was surprised that the range is used in the same way in both case. I would have expected one to be the inverse of the other.

Namely, we do:
- read: tensor = image * scale + bias
- write: image = tensor * scale + bias

I would have expected something like:
- read: tensor = image * scale + bias
- write: image = (tensor - bias )/ scale

What do you think?

Reply:
The 90 degree rotation was easy. We used the index of the width for the row and the index of the height for the column whereas that should be the other way around.

With both the scale stuff and that fix, the images look similar.

Reply:
Nice catch @qcolombet!